GitHubClient.py

```diff
@@ -0,0 +1,355 @@
+        issue_number = issue_request.json()['number'] if request_status in [200, 201] else None
+
+        # Check if issue should be added to a project now it exists.
+        if issue_number and self.project:
+            project_id = self._get_project_id(self.project)
+            if project_id:
+                owner, repo = self.repo.split('/')
+                issue_id = self._get_issue_global_id(owner, repo, issue_number)
+                if issue_id:
+                    self._add_issue_to_project(issue_id, project_id)
+
+        return request_status, issue_number
+
+    def close_issue(self, issue):
+        """Check to see if this issue can be found on GitHub and if so close it."""
+        issue_number = None
+        if issue.issue_number:
+            # If URL insertion is enabled.
+            issue_number = issue.issue_number
+        else:
+            # Try simple matching.
+            matched = 0
+            for existing_issue in self.existing_issues:
+                if existing_issue['title'] == issue.title:
+                    matched += 1
+                    # If there are multiple issues with similar titles, don't try and close any.
+                    if matched > 1:
+                        print(f'Skipping issue (multiple matches)')
+                        break
+                    issue_number = existing_issue['number']
+        if issue_number:
+            update_issue_url = f'{self.issues_url}/{issue_number}'
+            body = {'state': 'closed'}
+            requests.patch(update_issue_url, headers=self.issue_headers, json=body)
+            request_status = self._comment_issue(issue_number, f'Closed in {self.sha}.')
+
+            # Update the description if this is a PR.
+            if os.getenv('GITHUB_EVENT_NAME') == 'pull_request':
+                pr_number = os.getenv('PR_NUMBER')
+                if pr_number:
+                    request_status = self._update_pr_body(pr_number, body)
+            return request_status
+        return None
+
+    def _update_pr_body(self, pr_number, issue_number):
+        """Add a close message for an issue to a PR."""
+        pr_url = f'{self.repos_url}{self.repo}/pulls/{pr_number}'
+        pr_request = requests.get(pr_url, headers=self.issue_headers)
+        if pr_request.status_code == 200:
+            pr_body = pr_request.json()['body']
+            close_message = f'Closes #{issue_number}'
+            if close_message not in pr_body:
+                updated_pr_body = f'{pr_body}\n\n{close_message}' if pr_body.strip() else close_message
+                body = {'body': updated_pr_body}
+                pr_update_request = requests.patch(pr_url, headers=self.issue_headers, json=body)
+                return pr_update_request.status_code
+        return pr_request.status_code
+
+    def get_issue_url(self, new_issue_number):
+        return f'Issue URL: {self.line_base_url}{self.repo}/issues/{new_issue_number}'
```

Comment:
The syntax of this URL is different between GitHub and GitLab, so I pulled into the class (as a new method) rather than leaving it in `main`.